packager/packager.cc

```diff
@@ -585,6 +606,62 @@ class EnforcePackagePrefix final {
         return original;
     }
 
+    ast::ExpressionPtr postTransformAssign(core::Context ctx, ast::ExpressionPtr original) {
+        if (errorCount > 0) {
+            errorCount--;
+        }
+        return original;
+    }
+
+    ast::ExpressionPtr preTransformMethodDef(core::Context ctx, ast::ExpressionPtr original) {
+        if (errorCount > 0) {
+            errorCount++;
+            return original;
+        }
+        auto &def = ast::cast_tree_nonnull<ast::MethodDef>(original);
+        checkBehaviorLoc(ctx, def.declLoc);
+        return original;
+    }
+
+    ast::ExpressionPtr postTransformMethodDef(core::Context ctx, ast::ExpressionPtr original) {
+        if (errorCount > 0) {
+            errorCount--;
+        }
+        return original;
+    }
+
+    ast::ExpressionPtr preTransformSend(core::Context ctx, ast::ExpressionPtr original) {
+        if (errorCount > 0) {
+            errorCount++;
+            return original;
+        }
+        checkBehaviorLoc(ctx, original.loc());
+        return original;
+    }
+
+    ast::ExpressionPtr postTransformSend(core::Context ctx, ast::ExpressionPtr original) {
+        if (errorCount > 0) {
+            errorCount--;
+        }
+        return original;
+    }
+
+    void checkBehaviorLoc(core::Context ctx, core::LocOffsets loc) {
+        ENFORCE(errorCount == 0);
+        if (rootConsts > 0 || nameParts.empty()) {
+            return;
+        }
+        auto &pkgName = requiredNamespace(ctx.state);
+        if (!isPrefix(pkgName, nameParts)) {
+            errorCount++;
```

Comment:
This should hold here, right?

```suggestion
            ENFORCE(errorCount == 0);
            errorCount++;
```